HC/WrapperGen/WrapperGen.cpp

```diff
@@ -521,15 +534,13 @@ struct StringFinder
           out << "  static accelerator_view av = accelerator().get_default_view();\n";
           out << "  _lp.av = &av;\n";
           out << "}\n\n";
-          out << "completion_future cf = parallel_for_each(*(_lp.av),extent<3>(_lp.gridDim.z*_lp.groupDim.z,_lp.gridDim.y*_lp.groupDim.y,_lp.gridDim.x*_lp.groupDim.x).tile_with_dynamic(_lp.groupDim.z, _lp.groupDim.y, _lp.groupDim.x, _lp.groupMemBytes), \n"
+          out << "completion_future cf = parallel_for_each(*(_lp.av),extent<3>(_lp.grid_dim.z*_lp.group_dim.z,_lp.grid_dim.y*_lp.group_dim.y,_lp.grid_dim.x*_lp.group_dim.x).tile_with_dynamic(_lp.group_dim.z, _lp.group_dim.y, _lp.group_dim.x, _lp.dynamic_group_mem_bytes), \n"
               << func->getFunctorName()
               << "(";
           func->printArgsAsArguments(out);
           out << "));\n\n"
               << "if(_lp.cf)\n"
               << "  *(_lp.cf) = cf;\n"
-              << "else\n"
-              << "  cf.wait();\n"
```

Comment:
Right – now both grid_launch and PFE are always asynchronous, and user applies the desired synchronization through one of the other available methods (barrier bits, completion_future, accelerator_view::wait, etc) depending on the requirements of the application.


From: Wen-Heng (Jack) Chung [mailto:notifications@github.com]
Sent: Wednesday, June 22, 2016 8:08 PM
To: RadeonOpenCompute/hcc <hcc@noreply.github.com>
Cc: Sander, Ben <ben.sander@amd.com>; Author <author@noreply.github.com>
Subject: Re: [RadeonOpenCompute/hcc] Grid launch 20 a (#80)


In HC/WrapperGen/WrapperGen.cpp<https://github.com/RadeonOpenCompute/hcc/pull/80#discussion_r68162119>:

>                << func->getFunctorName()

>                << "(";

>            func->printArgsAsArguments(out);

>            out << "));\n\n"

>                << "if(_lp.cf)\n"

>                << "  *(_lp.cf) = cf;\n"

> -              << "else\n"

> -              << "  cf.wait();\n"

this make grid_launch kernels truly async, no matter a completion_future is supplied. I like having a consistent semantic.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/RadeonOpenCompute/hcc/pull/80/files/758b7f57d57a3a927dd454b7d8011318ecb129f0#r68162119>, or mute the thread<https://github.com/notifications/unsubscribe/ACYSAnpVdPxtaLwVpQdJnQjZe-54fiD0ks5qOdxUgaJpZM4I8LAJ>.